HelpHunterBE/Controllers/SearchController.cs

```diff
@@ -0,0 +1,169 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using Npgsql;
+using GeoCoordinatePortable;
+using HelpHunterBE.Utils;
+
+namespace HelpHunterBE.Controllers
+{
+    [Route("api")]
+    [ApiController]
+    public class SearchController : ControllerBase
+    {
+        private readonly IConfiguration _configuration;
+
+        public SearchController(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        [HttpPost("search")]
+        [Authorize]
+        public IActionResult Search([FromBody] SearchCriteria criteria)
+        {
+            try
+            {
+                using (var connection = new NpgsqlConnection(_configuration.GetConnectionString("Postgres")))
+                {
+                    connection.Open();
+                    // Construct the base SQL query
+                    string sqlQuery = @"SELECT
+                                        u.user_id,
+                                        u.full_name,
+                                        u.location,
+                                        u.location_coordinates_x,
+                                        u.location_coordinates_y,
+                                        s.service_id,
+                                        s.service_name,
+                                        avs.max_price,
+                                        avs.min_price,
+                                        avs.operating_mode,
+                                        c.category_name,
+                                        c.category_id,
+                                        AVG(r.rating) AS average_rating
+                                        FROM
+                                        Users u
+                                        JOIN
+                                        Available_Services avs ON u.user_id = avs.user_id
+                                        JOIN
+                                        Services s ON avs.service_id = s.service_id
+                                        JOIN
+                                        Categories c ON avs.category_id = c.category_id
+                                        LEFT JOIN
+                                        Ratings r ON u.user_id = r.user_id
+                                        WHERE 1 = 1";
```

Comment:
```suggestion
                                        WHERE 1 = 1";
```

To wygląda jak atak XSS z ostatnich labów xD Po co takie coś?

Reply:
to daje mi możliwość później doklejania bezproblemowo kolejnych elementów które się zaczynają od "AND...." 
BO jeśli mam AND a nie mam WHERE no to się query nie wykona.
WHERE 1=1 daje mi zwrotkę wszystkich wyników więc nie wpływa na wykonanie.